lib/ohai/plugins/ec2.rb

```diff
@@ -122,11 +122,17 @@ def looks_like_ec2?
       fetch_metadata.each do |k, v|
         # fetch_metadata returns IAM security credentials, including the IAM user's
         # secret access key. We'd rather not have ohai send this information
-        # to the server.
-        # http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AESDG-chapter-instancedata.html#instancedata-data-categories
-        next if k == "iam" && !hint?("iam")
-
-        ec2[k] = v
+        # to the server. If the instance is associated with an IAM role we grab
+        # only the "info" key and the IAM role name.
+        # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-categories.html
+        if k == "iam" && !hint?("iam")
+          ec2[:iam] = v.select { |key, value| key == "info" }
```

Comment:
I guess you did this for futureness? Less efficient than `ec2[:iam]['info'] = v['info']`, but I'm fine with this.

Reply:
yeah mostly for futureness and slightly because i thought it was more elegant than having the 'info' string in 2 places and also having to also do ec2[:iam] = {} first. If info isn't there for some reason select will still return an empty hash.